libr/core/bin.c

```diff
@@ -394,20 +394,36 @@ static const char* get_compile_time(Sdb *binFileSdb) {
 	return timeDateStamp_string;
 }
 
+static int is_executable (RBinObject *obj) {
+	RListIter *it;
+	RBinSection* sec;
+	if (obj) {
+		if (obj->info && obj->info->arch)
+			return true;
+		r_list_foreach (obj->sections, it, sec) {
+			if (R_BIN_SCN_EXECUTABLE & sec->srwx)
+				return true;
+		}
+	}
+	return false;
+}
+
 static int bin_info(RCore *r, int mode) {
 	int i, j, v;
 	char str[R_FLAG_NAME_SIZE];
 	char size_str[32];
 	char baddr_str[32];
 	RBinInfo *info = r_bin_get_info (r->bin);
 	RBinFile *binfile = r_core_bin_cur (r);
+	RBinObject *obj = r_bin_cur_object (r->bin);
 	const char *compiled = NULL;
+	bool havecode;
 
-	if (!binfile || !info) {
+	if (!binfile || !info || !obj) {
 		if (mode & R_CORE_BIN_JSON) r_cons_printf ("{}");
 		return false;
 	}
-
+	havecode = is_executable (obj) | (obj->entries != NULL);
```

Comment:
that is why here i check if the file contains executable section or it has an entry point